libraries/node-core-library/src/Import.ts

```diff
@@ -83,4 +128,105 @@ export class Import {
     const importLazyLocal: (moduleName: string) => unknown = importLazy(require);
     return importLazyLocal(moduleName);
   }
+
+  /**
+   * Resolves a path in a package, relative to another path.
+   */
+  public static resolve(options: IImportResolveOptions): string {
+    const { resolvePath } = options;
+
+    if (nodeJsPath.isAbsolute(resolvePath)) {
+      return resolvePath;
+    }
+
+    let normalizedRootPath: string = FileSystem.getRealPath(options.baseFolderPath);
+
+    if (resolvePath.startsWith('.')) {
+      // This looks like a conventional relative path
+      return nodeJsPath.resolve(normalizedRootPath, resolvePath);
+    }
+
+    normalizedRootPath =
+      PackageJsonLookup.instance.tryGetPackageFolderFor(normalizedRootPath) || normalizedRootPath;
+
+    let slashAfterPackageNameIndex: number;
+    if (resolvePath.startsWith('@')) {
+      // This looks like a scoped package name
+      slashAfterPackageNameIndex = resolvePath.indexOf('/', resolvePath.indexOf('/') + 1);
+    } else {
+      slashAfterPackageNameIndex = resolvePath.indexOf('/');
+    }
+
+    let packageName: string;
+    let pathInsidePackage: string | undefined;
+    if (slashAfterPackageNameIndex === -1) {
+      // This looks like a package name without a path
+      packageName = resolvePath;
+    } else {
+      packageName = resolvePath.substr(0, slashAfterPackageNameIndex);
+      pathInsidePackage = resolvePath.substr(slashAfterPackageNameIndex + 1);
```

Comment:
It seems like `require.sync()` already implements all these parsing features itself.  It almost exactly simulates Node.js `require()` behavior.